src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,175 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+
+    private class Node<E> {
+        private E item;
+        private Node<E> next;
+        private Node<E> prev;
+
+        Node(Node<E> prev, E element, Node<E> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (first == null) {
+            first = new Node<>(null, value, null);
+            return true;
+        }
+        Node<T> element = first;
+        while (element.next != null) {
+            element = element.next;
+        }
+        Node<T> newNode = new Node<T>(element, value, null);
+        element.next = newNode;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index);
+        if (first == null) {
+            add(value);
+            return;
+        }
+        Node<T> currentNode = getNodeByIndex(index);
 
+        if (currentNode == null) {
+            Node<T> prevCurrent = getNodeByIndex(index - 1);
+            if (prevCurrent == null) {
+                throw new IndexOutOfBoundsException("Index is bigger than list size");
+            }
+            add(value);
+            return;
+        }
+
+        Node<T> prevCurrent = currentNode.prev;
+        Node<T> newNode = new Node<>(prevCurrent, value, currentNode);
+        if (prevCurrent != null) {
+            prevCurrent.next = newNode;
+        } else {
+            first = newNode;
+        }
+        currentNode.prev = newNode;
```

Comment:
Creating a new node and linking it to the previous and next is better to make a separate method.   
Something like this - addByIndex(T value, int index)